### ISSUE TITLE
Support updating repository url

### ISSUE THREAD

**Issue by mattiasb** [acch/ansible-scale#7]
When using the `ansible-scale-gui` role I accidentally set wrong repository urls for `scale_install_repository_url` and `scale_install_zimon_repository_url` which made ansible fail mid through applying the role. Changing the repository urls after the fact didn't work and I had to manually remove the bad repo files under `/etc/yum.repos.d/` before reapplying the role with the correct url's set. You probably need to use the `ini` module to update an existing repo file. 

While this was found while using the `ansible-scale-gui` role the same problem should arise here as well, so posting it here.

As a sidenote this is what I ended up setting:

```yaml
scale_install_repository_url: >-
  file:///usr/lpp/mmfs/{{ scale_version }}/gpfs_rpms/

scale_install_zimon_repository_url: >-
  file:///usr/lpp/mmfs/{{ scale_version }}/zimon_rpms/rhel7/

```

**Comment by acch** [comment]
Thanks - that's a good catch!

From the Ansible yum_repository docs:
> If you wish to update an existing repository definition use ini_file instead.

https://docs.ansible.com/ansible/latest/modules/ini_file_module.html#ini-file-module

**Comment by acch** [comment]
I'm sorry that I haven't been able to invest time in this, yet. It's this time of the year again...

**Comment by acch** [comment]
I was looking at this issue again just now, and I can't seem to be able to reproduce the behavior. Looks like `yum_repository` will now update existing files if the baseurl changes.

Ansible version: 2.9.14
Python version: 3.8.6
Managed host: RHEL 7.8

This can be verified with a quick test:

```yaml
- hosts: node1
  tasks:

    - name: test | Clean existing YUM repository
      file:
        path: /etc/yum.repos.d/spectrum-scale-gpfs.repo
        state: absent

    - command: ls -l /etc/yum.repos.d/

    - name: test | Configure new YUM repository
      yum_repository:
        name: spectrum-scale-gpfs
        description: IBM Spectrum Scale (GPFS)
        baseurl: http://infraserv/yum/gpfs/5.0.5.1/gpfs_rpms
        state: present

    - command: grep baseurl /etc/yum.repos.d/spectrum-scale-gpfs.repo

    - name: test | Modify existing YUM repository
      yum_repository:
        name: spectrum-scale-gpfs
        description: IBM Spectrum Scale (GPFS)
        baseurl: http://infraserv/yum/gpfs/5.0.5.3/gpfs_rpms
        state: present

    - command: grep baseurl /etc/yum.repos.d/spectrum-scale-gpfs.repo
```

The output of the first `grep` is:
`baseurl = http://infraserv/yum/gpfs/5.0.5.1/gpfs_rpms`

The output of the second `grep` is:
`baseurl = http://infraserv/yum/gpfs/5.0.5.3/gpfs_rpms` (version number has changed)

Hence, I guess that there were changes to the `yum_repository` module and it now works as expected (modifies existing repos).

Are you able to reproduce the issue with Ansible 2.9?

**Comment by mattiasb** [comment]
@acch I posted this bug during an education session regarding Spectrum Scale, part of the session was to install Spectrum Scale over a couple of virtual machines and I decided to try using this role instead of using the manual installation method described in the course. 

In short: I don't have access to Spectrum Scale right now so can't really help. I'm sorry. If this works on your end (and the test looks sound to me) I'd assume this got fixed in Ansible itself. :)
